docs/Permissions/Onchain-Permissioning.md

```diff
@@ -0,0 +1,191 @@
+
+    * `PANTHEON_NODE_PERM_ENDPOINT` - required only if your node is not using the default JSON-RPC host and port (`http://127.0.0.1:8545`). 
+       Set to JSON-RPC host and port. 
+
+    !!! note
+        If your network is not configured as a [free gas network](../Configuring-Pantheon/FreeGas.md), the account used to 
+        interact with the permissioning contracts must have a balance. 
+
+1. Clone the `permissioning-smart-contracts` repository: 
+
+    ```bash 
+    git clone https://github.com/PegaSysEng/permissioning-smart-contracts.git
+    ```
+
+1. Change into the `permissioning-smart-contracts` directory and run: 
+
+    ```bash
+    npm install
+    ```
+
+## Deploy Rules Contract 
+
+In the `permissioning-smart-contracts` directory, deploy the Rules contract: 
+
+```bash
+truffle migrate --reset
+```
+
+The Rules contract is deployed and the Ingress contract updated with the name and version of the Rules contract. 
+
+!!! important 
+    The account that deploys the Rules contract is automatically an [Admin account](#add-and-remove-admin-accounts). 
+
+## Add and Remove Nodes from the Whitelist 
+
+!!! note  
+    Only [Admin accounts](#add-and-remove-admin-accounts) can add or remove nodes from the whitelist. 
+
+To add or remove nodes: 
+
+1. In the `permissioning-smart-contracts` directory, run the Truffle console: 
+
+    ```bash
+    truffle console 
+    ```
+
+1. Open https://permissioning-tools.pegasys.tech/
+
+1. Enter the [enode URL](../Configuring-Pantheon/Node-Keys.md#enode-url) of the node to be added or removed
+and click the *Process* button. 
+
+1. Click the *Copy to clipboard* button for _Add Enode_ or _Remove Enode_. 
+
+1. Paste the copied command into the Truffle Console. 
+
+   When the transaction is included in a block, the transaction receipt is displayed. 
+
+!!! important 
+    When starting the network, the first node must add itself to the whitelist.  
+
+Added nodes do not attempt to reconnect until peer discovery restarts.  To add a whitelisted node as a peer
+without waiting for peer discovery to restart, use [`admin_addPeer`](../Reference/JSON-RPC-API-Methods.md#admin_addpeer). 
+
+## Display Nodes Whitelist 
+ 
+To display the nodes whitelist, paste the following into the Truffle Console: 
+
+```bash
+Rules.deployed().then(function(instance) {instance.getKeyCount().then(function(txCount) {console.log("size of whitelist: " + txCount); var i=txCount; while(i>=0) {instance.getWhitelistKey(i--).then(function(tx) {console.log(tx)})}});});
+```
+
+## Start Other Network Nodes 
+
+For participating nodes that are not going to add or remove nodes from the whitelist:
+
+1. Start the nodes with permissioning contract options: 
+
+    * [--permissions-nodes-contract-enabled](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-enabled)
+     to enable onhcain permissioning
+    
+    * [--permissions-nodes-contract-address](../Reference/Pantheon-CLI-Syntax.md#permissions-nodes-contract-address)
```

Comment:
Don't we have a default value for this if we alway advertise using `0x0000000000000000000000000000000000009999` ?